src/Chainweb/Pact/PactService.hs

```diff
@@ -446,18 +447,21 @@ applyPactCmds
     -> Vector (P.Command PayloadWithText)
     -> MinerInfo
     -> PactServiceM (Vector HashCommandResult)
-applyPactCmds isGenesis env' cmds miner = V.mapM f cmds
+applyPactCmds isGenesis env' cmds miner =
+    fmap (\(T2 v _) -> v) $ V.foldM f (T2 mempty mempty) cmds
```

Comment:
Turns out:
```haskell
instance (Monoid a, Monoid b) => Monoid (T2 a b) where
  mempty = T2 mempty mempty
```

Reply:
You might also consider `<$>` (hlint would warn).

Reply:
Too bad `strict-tuple` doesn't have simple accessor functions eh?

Reply:
Ah it does have a monoid instance! Okay, that's tricky though, because that's a pretty horrible monoid 😄 (no shortcircuiting!) . Tell you what. I'll write a module for `Data.Tuple.Strict` in `src` and write the convenience functions for us.

Reply:
Just put it in `Chainweb.Util`.

Reply:
Sounds good